### PR TITLE
[develop] CI platform directories

### PR DESCRIPTION
**Pull request by scottarnette**
## Change Description

- Restructured `.cicd` files to support new platform types separated into distinct directories moving forward.
- Removed use of `BUILDKITE_AGENT_ACCESS_TOKEN`. It was unneeded.
- Removed latest tag in Contracts Builder. This would flip-flop between being a image based off different branches and was unreliable.
- Updated `scripts/serial-test.sh` and `scripts/long-running-test.sh` to not start mongod if it does not exist.
  - Added bonus of no longer needing to install Mongo dependencies during Mac builds on Travis, speeding that build time up.
- Updated Ubuntu 18 unpinned to use Clang-7 to match AmazonLinux2 unpinned.

See:
https://buildkite.com/EOSIO/eosio/builds/18382 | Pinned
https://buildkite.com/EOSIO/eosio-build-unpinned/builds/813 | Unpinned
https://buildkite.com/EOSIO/eosio-build-unpinned/builds/814 | Canceled build showing that Anka correctly generates images.

## Consensus Changes
- [ ] Consensus Changes

## API Changes
- [ ] API Changes

## Documentation Additions
- [ ] Documentation Additions

### CODE REVIEW

File: .cicd/installation-build.sh [hunk 1]
```diff
@@ -7,11 +7,9 @@ if [[ $BUILDKITE_BRANCH =~ ^release/[0-9].[0-9]+.x$ || $BUILDKITE_BRANCH =~ ^mas
     export CONTRACTS_BUILDER_TAG="eosio/ci-contracts-builder:base-ubuntu-18.04"
     export ARGS="--name ci-contracts-builder-$BUILDKITE_COMMIT --init -v $(pwd):$MOUNTED_DIR"
     $CICD_DIR/build.sh
-    docker commit ci-contracts-builder-$BUILDKITE_COMMIT $CONTRACTS_BUILDER_TAG-latest
```
Comment (zreyn) [inline .cicd/installation-build.sh]: Good riddance to the latest tag!

File: .cicd/build.sh [hunk 2]
```diff
@@ -22,18 +21,17 @@ if [[ "$(uname)" == 'Darwin' ]]; then
 else # Linux
     CMAKE_EXTRAS="$CMAKE_EXTRAS -DBUILD_MONGO_DB_PLUGIN=true"
     ARGS=${ARGS:-"--rm --init -v $(pwd):$MOUNTED_DIR"}
-    . $HELPERS_DIR/file-hash.sh $CICD_DIR/platforms/$IMAGE_TAG.dockerfile
     PRE_COMMANDS="cd $MOUNTED_DIR/build"
     # PRE_COMMANDS: Executed pre-cmake
     # CMAKE_EXTRAS: Executed within and right before the cmake path (cmake CMAKE_EXTRAS ..)
     [[ ! "$IMAGE_TAG" =~ 'unpinned' ]] && CMAKE_EXTRAS="$CMAKE_EXTRAS -DCMAKE_TOOLCHAIN_FILE=$MOUNTED_DIR/.cicd/helpers/clang.make -DCMAKE_CXX_COMPILER_LAUNCHER=ccache"
-    if [[ "$IMAGE_TAG" == 'amazon_linux-2' ]]; then
+    if [[ $IMAGE_TAG == 'amazon_linux-2-pinned' ]]; then
```
[the remark this answers is not in the document]
Comment (scottarnette) [reply .cicd/build.sh]: `PRE_COMMANDS` are used across all of the images for various things. For Travis, they set the correct location of ccache. For unpinned CentOS builds, this enables devtooset-7 and python 3.

File: scripts/serial-test.sh [hunk 3]
```diff
@@ -16,10 +16,13 @@ if [[ ! -z "$(pgrep mongod)" ]]; then
     echo "+++ $([[ "$BUILDKITE" == 'true' ]] && echo ':leaves: ')Killing old MongoDB"
     $(pgrep mongod | xargs kill -9) || :
 fi
-echo "+++ $([[ "$BUILDKITE" == 'true' ]] && echo ':leaves: ')Starting new MongoDB"
-[[ ! -d ~/data/mongodb && ! -d mongodata ]] && mkdir mongodata
-echo "$ mongod --fork --logpath $(pwd)/mongod.log $([[ -d ~/data/mongodb ]] && echo '--dbpath ~/data/mongodb' || echo "--dbpath $(pwd)/mongodata") $(if [[ -f ~/etc/mongod.conf ]]; then echo '-f ~/etc/mongod.conf'; elif [[ -f /usr/local/etc/mongod.conf ]]; then echo '-f /usr/local/etc/mongod.conf'; fi)"
-eval mongod --fork --logpath $(pwd)/mongod.log $([[ -d ~/data/mongodb ]] && echo '--dbpath ~/data/mongodb' || echo "--dbpath $(pwd)/mongodata") $(if [[ -f ~/etc/mongod.conf ]]; then echo '-f ~/etc/mongod.conf'; elif [[ -f /usr/local/etc/mongod.conf ]]; then echo '-f /usr/local/etc/mongod.conf'; fi)
+# do not start mongod if it does not exist
+if [[ -x $(command -v mongod) ]]; then
```
[the remark this answers is not in the document]
Comment (scottarnette) [reply scripts/serial-test.sh]: 🎈

File: .cicd/installation-build.sh (hunk 1 above)
Comment (scottarnette) [reply .cicd/installation-build.sh]: 🎉

**Pull request merged**